homeassistant/components/light/yeelight.py

```diff
@@ -286,13 +328,428 @@ def set_flash(self, flash) -> None:
             except BulbException as ex:
                 _LOGGER.error("Unable to set flash: %s", ex)
 
+    @_cmd
+    def set_effect(self, effect) -> None:
+        """Activate effect."""
+        if effect:
+            from yeelight import (RGBTransition, TemperatureTransition,
+                                  SleepTransition, Flow, BulbException)
+            transition = int(self.config[CONF_TRANSITION])
+            if effect == EFFECT_ALARM:
```

Comment:
I think that the logic for the effects should be located in the `yeelight` module itself. This way the effects can be maintained at one place.